kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -189,40 +240,32 @@ def after_catalog_created(self, catalog):
         )
 
 
-def _is_known_ci_env(known_ci_env_var_keys=KNOWN_CI_ENV_VAR_KEYS):
+def _is_known_ci_env(known_ci_env_var_keys: set[str]):
     # Most CI tools will set the CI environment variable to true
     if os.getenv("CI") == "true":
         return True
     # Not all CI tools follow this convention, we can check through those that don't
     return any(os.getenv(key) for key in known_ci_env_var_keys)
 
 
-def _get_project_properties(user_uuid: str, project_path: Path) -> dict:
-    hashed_package_name = _hash(str(PACKAGE_NAME)) if PACKAGE_NAME else "undefined"
+def _get_project_properties(user_uuid: str, pyproject_path: Path) -> dict:
+    project_uuid = _get_or_create_project_uuid(pyproject_path)
+    package_name = PACKAGE_NAME or UNDEFINED_PACKAGE_NAME
+    hashed_project_uuid = (
+        _hash(f"{project_uuid}{package_name}") if project_uuid is not None else None
+    )
+
     properties = {
         "username": user_uuid,
-        "package_name": hashed_package_name,
+        "project_uuid": hashed_project_uuid,
```

Comment:
Nitpick: Shall we call this `project_id`? (since it's not going to be an UUID, but rather a hash of a combination of UUID + something else)

```suggestion
        "project_id": hashed_project_id,
```